### ISSUE TITLE
Godot server build cannot load PNG images

### ISSUE THREAD

**Issue by and3rson** [godotengine/godot#28747]
**Godot version:**
v3.1-stable

**OS/device including version:**
ArchLinux rolling

**Issue description:**
Godot server build does not load PNG images for some reason. On the other hand, using anything other than PNG or running X11 build works fine.

**Steps to reproduce:**
1. Create `TextureRect` and set texture to some PNG image
2. Run project with Godot server build
3. Observe the following errors:

```
ERROR: create: Expected data size of 262144 bytes in Image::create(), got instead 349524 bytes.
   At: core/image.cpp:1469.                                                                                           
ERROR: create: Expected data size of 262144 bytes in Image::create(), got instead 349532 bytes.
   At: core/image.cpp:1469.     
ERROR: create: Expected data size of 262144 bytes in Image::create(), got instead 349532 bytes.
   At: core/image.cpp:1469.                                                                                           
ERROR: create: Expected data size of 2560000 bytes in Image::create(), got instead 3413260 bytes.
   At: core/image.cpp:1469.     
ERROR: create: Expected data size of 2560000 bytes in Image::create(), got instead 3413260 bytes.
   At: core/image.cpp:1469.                                                                                           
ERROR: create: Expected data size of 3240000 bytes in Image::create(), got instead 4319376 bytes.
   At: core/image.cpp:1469.     
ERROR: create: Expected data size of 2890000 bytes in Image::create(), got instead 3852020 bytes.
   At: core/image.cpp:1469.                                                                                           
ERROR: create: Expected data size of 409600 bytes in Image::create(), got instead 546120 bytes.
   At: core/image.cpp:1469.     
ERROR: create: Expected data size of 2560000 bytes in Image::create(), got instead 3413260 bytes.
   At: core/image.cpp:1469.                                                                                           
```

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in [Godot 3.2.3 rc3](https://godotengine.org/article/release-candidate-3-2-3-rc-3) or any later release?

If yes, please ensure that an up-to-date Minimal Reproduction Project (MRP) is included in this report (a MRP is a zipped Godot project with the minimal elements necessary to reliably trigger the bug). You can upload ZIP files in an issue comment with a drag and drop.

**Comment by cesarizu** [comment]
[PNG Import Error.zip](https://github.com/godotengine/godot/files/5543108/PNG.Import.Error.zip)

I've attached a minimal sample project that presents this problem.

There are two images, the exact same file, but with different import options. The difference is just enabling mipmap generation. When I enable that, the image is broken and not imported. When I disable that, the image is imported correctly.

This is happening for me on 3.2.3 final. This is a log of the problem:

```
 Expected data size of 5592404 bytes in Image::create(), got instead 5505023 bytes.
 drivers/gles2/rasterizer_storage_gles2.cpp:706 - Condition "!read.ptr()" is true.
 drivers/gles2/rasterizer_storage_gles2.cpp:706 - Condition "!read.ptr()" is true.
 drivers/gles2/rasterizer_storage_gles2.cpp:814 - Condition "texture->data_size == 0 && !texture->render_target" is true. Returned: Ref<Image>()
 drivers/gles2/rasterizer_storage_gles2.cpp:814 - Condition "texture->data_size == 0 && !texture->render_target" is true. Returned: Ref<Image>()
 drivers/gles2/rasterizer_storage_gles2.cpp:814 - Condition "texture->data_size == 0 && !texture->render_target" is true. Returned: Ref<Image>()
 drivers/gles2/rasterizer_storage_gles2.cpp:814 - Condition "texture->data_size == 0 && !texture->render_target" is true. Returned: Ref<Image>()
...
```

**Comment by Zylann** [comment]
Tested in Godot 3.2.3, on Windows 10 64 bits OpenGL ES 2.0 Renderer: GeForce GTX 1060 6GB/PCIe/SSE2
And I can reproduce the problem.

Then I tested with the same setup in Godot 3.4 rc1 (after having deleted the `.import` folder), I only get this error:
```
ERROR: Another resource is loaded from path 'res://a.png' (possible cyclic resource inclusion).
   at: (core/resource.cpp:79)
```
But the two PNGs seem to import correctly.

After retrying again deleting the import folder, the project re-imports correctly without errors.
